providers/firestore.js

```diff
@@ -0,0 +1,183 @@
+	 * @returns {Promise<Array>}
+	 */
+	getKeys(table) {
+		return this.db.collection(table).get().then(snaps => snaps.docs.map(snap => snap.id));
+	}
+	/**
+	 * Returns a particular Document from the table/Collection.
+	 * @param {string} table Name of the collection/Table to get the value from
+	 * @param {string} id Name of the Document/Key to be returned
+	 * @returns {Promise<Any>}
+	 */
+	get(table, id) {
+		return this.db.collection(table).doc(id).get();
+	}
+
+	/**
+	 * Checks if a collection has a particular Document/Key
+	 * @param {string} table Name of the collection/Table
+	 * @param {string} id Name of the Document/Key
+	 * @returns {Promise<boolean>}
+	 */
+	has(table, id) {
+		return this.db.collection(table).doc(id).get().then(data => data.exists);
+	}
+
+	/**
+	 * Creates a new Document/Field in a collection.
+	 * @param {string} table Name of the collection/table
+	 * @param {string} id Name of they Docment/Key
+	 * @param {(ConfigurationUpdateResultEntry[] | [string, any][] | Object<string, *>)} doc the object you want to insert in the table.
+	 * @returns {Promise<DocumentObject>}
+	 */
+	async create(table, id, doc = {}) {
+		return await this.db.collection(table).doc(id).set(this.parseUpdateInput(doc));
+	}
+
+	/**
+	 * Updates the values of a Document/Field
+	 * @param {string} table Name of the Collection that contains the Document
+	 * @param {string} id  Name of the Document/field
+	 * @param {(ConfigurationUpdateResultEntry[] | [string, any][] | Object<string, *>)} doc the object you want to insert in the table.
+	 * @returns {Promise<WriteResult>}
+	 */
+	async update(table, id, doc) {
+		return await this.db.collection(table).doc(id).set(this.parseUpdateInput(doc), { merge: true });
+	}
+
+	/**
+	 * Deleted a particular Document/Field from the Collection/Table
+	 * @param {string} table Name of the collection/table
+	 * @param {string} id Name of the Document/Key to delete
+	 * @returns {Promise<WriteResult>}
+	 */
+	async delete(table, id) {
+		return await this.db.collection(table).doc(id).delete();
+	}
+	/**
+	 * Replaces a Document with a new Document specified by the user *
+	 * @param {string} table Name of the Collection
+	 * @param {Object} id The Filter used to select the document to update
+	 * @param {Object} [doc={}] The Document that replaces the matching document
+	 * @returns {Promise}
+	 */
+	replace(...args) {
+		return this.create(...args);
+	}
+
+	/**
+	 * Update or insert a new value to all entries
+	 * @param {string} table Name of the Collectiom
+	 * @param {string} path The Field to be updated in al the Documents
+	 * @param {*} newValue The value to to updated in all the Documents
+	 */
+	async updateValue(table, path, newValue) {
+		const keys = await this.getKeys(table);
+
+		if (typeof path === 'object' && typeof newValue === 'undefined') {
+			await Promise.all(keys.map(doc => this.update(table, doc, path)));
```

Comment:
Should return or use else cases. With this, you'll make it always throw, same for removeValue.